src/onegov/search/mixins.py

```diff
@@ -134,7 +152,129 @@ def es_tags(self):
+        Searchable.drop_fts_column(session, schema, model)
+        Searchable.add_fts_column(session, schema, model)
+
+    @staticmethod
+    def drop_fts_column(session, schema, model, col_name='fts_idx'):
+        """
+        Drops the full text search column
+
+        :param session: db session
+        :param schema: schema the full text column shall be added
+        :param model: model to drop the index from
+        :param col_name: column name of fts index
+        :return: None
+        """
+        query = f"""
+            ALTER TABLE "{schema}".{model.__tablename__} DROP COLUMN IF EXISTS
+            {col_name};
+        """
+        session.execute(query)
+        session.execute("COMMIT")
+
+    @staticmethod
+    def add_fts_column(session, schema, model, col_name='fts_idx'):
+
+        """
+        This function is used for re-indexing and as migration step moving to
+        postgresql full text search (fts), OGC-508.
+
+        It adds a separate column for the tsvector to `schema`.`table`
+        creating a multilingual gin index on the columns/data defined per
+        model.
+
+        :param session: session
+        :param schema: schema name
+        :param model: model to add the index
+        :param col_name: column name of fts index
+        :return: None
+        """
+
+        # we build the tsvector expression for all supported languages based
+        # on this format
+        # to_tsvector('simple', '<tsvector_string_of_model') ||
+        # to_tsvector('german', '<tsvector_string_of_model') ||
+        # to_tsvector('french', '<tsvector_string_of_model') ||
+        # ...
+
+        tsvector_expression = \
+            Searchable.multi_language_tsvector_expression(
+                model.psql_tsvector_string(model))
+
+        query = f"""
+            ALTER TABLE "{schema}".{model.__tablename__}
+            ADD COLUMN IF NOT EXISTS {col_name} tsvector GENERATED ALWAYS AS
+            ({tsvector_expression}) STORED;
+        """
+        session.execute(query)
+        session.execute("COMMIT")
+
+    @staticmethod
+    def multi_language_tsvector_expression(tsvector_string):
+        return ' || '.join(
+            "to_tsvector('{}', {})".format(language, tsvector_string)
+            for language in get_fts_index_languages()
+        )
+
+    @staticmethod
+    def create_tsvector_string(*cols):
+        """
+        Creates tsvector string for columns
+        Doc reference:
+        https://www.postgresql.org/docs/current/textsearch-tables.html#TEXTSEARCH-TABLES-INDEX
+
+        :param cols: column names to be indexed
+        :return: tsvector string for multiple columns i.e. for columns title
+        and body: coalesce(\"title\", '') || ' ' || coalesce(\"body\", '')
+        Note that the '\"{}\"' escapes the column name in case it is a
+        keyword, see
+        http://www.postgresql.org/docs/current/static/sql-keywords-appendix.html
+        """
+
+        base = "'func.coalesce({}, '')'"
```

Comment:
Yes, that's still using string interpolation, the only difference is, that sqlalchemy will resolve coalesce to the generic function which can be replaced by driver specific version but the result is the same. You are still using the `||` operator instead of the generic concat function.

sqlalchemy will do its best to accept a string as a SQL expression and will try to substitute some symbols by name, but it adds a lot of overhead for one, and is very error prone and potentially dangerous in terms of properly escaping inputs (SQL Injection). Your API incorrectly assumes that it should it should build a tsvector string, but what it actually should build is a tsvector expression. I know there are some other places in OneGov where we use raw string expressions (particularly for JSONB access), but everything can be done with just function calls.